From: Christopher Obbard <chris.obbard@collabora.com>
Date: Tue, 19 Oct 2021 12:08:07 +0100
Subject: Ignore install error when slot is missing from bootorder

If a slot becomes non-bootable and looses the EFI binary,
the BIOS removes that slot from the bootorder variable and
re-enumerates the bootorder on each boot. This causes RAUC
to error out as it then thinks the slot doesn't exist
entirely.

This causes an upgrade to error out early when there isn't
actually an error with the system. So, let's ignore these
errors and continue with the upgrade.

Signed-off-by: Christopher Obbard <chris.obbard@collabora.com>
Forwarded: not-needed, hack for specific platform
---
 src/install.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/install.c b/src/install.c
index dc52e64..2110706 100644
--- a/src/install.c
+++ b/src/install.c
@@ -807,10 +807,8 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* bu
 		res = r_boot_set_state(dest_slot, FALSE, &ierror);
 
 		if (!res) {
-			g_set_error(error, R_INSTALL_ERROR, R_INSTALL_ERROR_MARK_NONBOOTABLE,
-					"Failed marking slot %s non-bootable: %s", dest_slot->name, ierror->message);
+			g_message("Failed marking slot %s non-bootable: %s", dest_slot->name, ierror->message);
 			g_clear_error(&ierror);
-			goto early_out;
 		}
 	}
 
